### PR TITLE
#184581184 create product

### PR DESCRIPTION
**Pull request by AbdulKhaliq59**
#### What does this PR do?
This PR Create/Add Product in seller's collection

#### Description of Task to be completed?
- [x] Create input validation
- [x] verify seller's token
- [x] Create endpoint for creating product
- [x] Swagger documentation for this this endpoint
- [x] test For this endpoint

#### How should this be manually tested?
- Clone this repo git clone [e-comm-team-amigos-bn]( https://github.com/atlp-rwanda/e-comm-team-amigos-bn)
- Switch your working directory cd e-comm-team-amigos-bn
- Create an .env file and update it with `.env-example`
- Install all the dependencies by running `npm install`
- Then run the project by typing `npm start`

#### Endpoints to be tested:
- `POST /user/login`
- `POST /user/otp`
- ` POST /products/create`

#### Any background context you want to provide?
- Have Node Js installed in your local machine

#### What are the relevant pivotal tracker/Trello stories?

Story ID: [#184581184](https://www.pivotaltracker.com/story/show/184581184)

#### Screenshots (if approp
screencapture-localhost-4000-docs-2023-03-16-20_06_49
Screen Shot 2023-03-16 at 20 11 53

### CODE REVIEW

**Review by GasanaJr — CHANGES_REQUESTED** [review]
Update the body of your Pull request to include screenshots of passing tests and Swagger documentation.

**Review by ericndungutse — APPROVED** [review]
LGTM

File: src/controllers/product.controller.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      expiryDate,
+      ec,
+    } = req.body;
+    const images = req.body.images || [];
+    const existingProduct = await models.Product.findOne({ where: { name } });
+    if (existingProduct) {
+      return res.status(409).json({
+        message: "Product already exists you can update that product instead",
+      });
+    }
```
Comment (Emile-SHUMBUSHO) [inline src/controllers/product.controller.js:63]: It's not a good practice to mix validation and create a product controller, better to handle it in middleware.

File: src/controllers/product.controller.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      available,
+      category,
+      bonus,
+      images,
+      expiryDate,
+      ec,
+    });
+    res.status(201).json(product);
+  } catch (error) {
+    res.status(400).json({ error });
```
Comment (Emile-SHUMBUSHO) [inline src/controllers/product.controller.js]: Inside catch block returning status code 400(bad request) does not convince the exact the cause of error better to use status code 500(internal server error)

File: src/config.js [hunk 3]
```diff
@@ -11,12 +11,12 @@ module.exports = {
...
   test: {
-    use_env_variable: '',
-    username: process.env.DB_USER,
-    password: process.env.DB_PASSWORD,
-    database: process.env.TEST_DB_NAME,
-    host: process.env.DB_HOST,
-    port: process.env.DB_PORT,
+    use_env_variable: 'DATABASE_URL',
+    // username: process.env.DB_USER,
+    // password: process.env.DB_PASSWORD,
```
Comment (nshutijonathan) [inline src/config.js]: remove commented out codes

**Review by nshutijonathan — COMMENTED** [review]
check a few comment I left

File: src/controllers/index.js
Comment (khevin2) [inline src/controllers/index.js:1]: Please remove console.log()

**Review by khevin2 — APPROVED** [review]

**Review by Emile-SHUMBUSHO — APPROVED** [review]

**Review by GasanaJr — APPROVED** [review]

**Review by Habinezajanvier — APPROVED** [review]
👍

**Pull request merged**